src/WebSdk/Publish/Tasks/Tasks/Kudu/KuduDeploy.cs

```diff
@@ -160,15 +157,18 @@ internal bool DeployZipFile(KuduConnectionInfo connectionInfo)
             return success && zipTask.Result;
         }
 
-        internal string CreateZipFile(string sourcePath)
+        internal string? CreateZipFile(string? sourcePath)
         {
             // Zip the files from PublishOutput path.
             string zipFileFullPath = Path.Combine(Path.GetTempPath(), string.Format("Publish{0}.zip", new Random().Next(int.MaxValue)));
             Log.LogMessage(Framework.MessageImportance.High, string.Format(Resources.KUDUDEPLOY_CopyingToTempLocation, zipFileFullPath));
 
             try
             {
-                System.IO.Compression.ZipFile.CreateFromDirectory(sourcePath, zipFileFullPath);
+                if (sourcePath is not null)
```

Comment:
This seems like it should be an error

Reply:
Ah, looking at this, I agree that is should go down the failure path. However, I'm not going to use the try-catch for that. I'll add a guard clause that does this check and returns null if `sourcePath` is null. Thanks for the suggestion!